### PR TITLE
Bump lower bound of build, test and run requirement to JDK 21

### PR DESCRIPTION
**Pull request by mbien**
new test and build range for NB 30+: JDK 21-26

some remain on 17 due to #7871

this is also updating tests / test data if needed

Some of the tests still use the SM to inspect changing state. I began reimplementing it using JFR, but then realized that we can't do that yet due to an chicken-egg problem. It would require bumping some modules to JDK 21 which we could do now, but it would cause CI failures due to #7871. Workaround is to set `-Djava.security.manager=allow` while trying to decide what do with the GraalVM job etc. Maybe we could exclude clusters from the GraalVM job.

~todo: update readme~

### CODE REVIEW

File: java/java.source.base/test/unit/src/org/netbeans/api/java/source/TreePathHandleTest.java [hunk 1]
```diff
@@ -170,29 +172,40 @@ public void test134457() throws Exception {
...
+                });
+                rs.startAsync();
+//                file2.getInputStream().read();  // check: commenting this in should fail the test
+                TreePathHandle.create(string, controller);
+                // TODO call directly post JDK 21 bump
+                RecordingStream.class.getMethod("stop").invoke(rs); // flushes stream
             }
         }, true);
+        
+        assertFalse("file was read", wasRead.get());
```
Comment (mbien) [inline java/java.source.base/test/unit/src/org/netbeans/api/java/source/TreePathHandleTest.java:208]: POC of how to replace `SecurityManager` based IO monitoring with `JFR`. This wont work in all cases since the `jdk.FileRead/Write` events are for read/write operations in the literal sense, not for file attribute access. (e.g the tests in the versioning module will likely require a different approach to replace SM usage there)

**Comment by mbien** [comment]
squashed a little bit, will squash more before merge

**Comment by mbien** [comment]
added commit for readme+launcher update

**Comment by mbien** [comment]
> constant aka '17' or '21' in named baseline_jdk

would be nice, we already have a constant for the JDK vendor. We probably have too many edge cases still but the way its set up we should be able to put the lower/mid/upper bounds into constants. Its just that sometimes we have releases when we want to test an extra version on a specific job or something like that.

> Minor thought might be to rename test.jms.flags which is somewhat misused here by that name.

yeah the flag isn't used as intended. But this hopefully only a temporary situation, until the tests are rewritten/removed. As permanent solution it would probably better to add an extra flag instead of renaming. Since some tests use it to reset the jms flags to empty.

>  And could we move the -J-Dbootstrap.disableJDKCheck=true under a TODO there too?

will either add TODOs or check if those added in https://github.com/apache/netbeans/commit/d0793acdbd4b9ddd764c5ce8484cf021e8c38a9d are still needed and maybe remove some.


thanks for the reviews! Will try to squash this somehow so that it still makes sense before merge.

**Comment by mbien** [comment]
> > And could we move the -J-Dbootstrap.disableJDKCheck=true under a TODO there too?
> 
> will either add TODOs or check if those added in [d0793ac](https://github.com/apache/netbeans/commit/d0793acdbd4b9ddd764c5ce8484cf021e8c38a9d) are still needed and maybe remove some.

looks like I cleaned some of them up already, there are only two remaining:
https://github.com/apache/netbeans/blob/bf4900bac4ed0d9222bfbcc56871b9b426ec3503/apisupport/apisupport.ant/test/unit/src/org/netbeans/modules/apisupport/project/suite/BuildZipDistributionTest.java#L236
https://github.com/apache/netbeans/blob/bf4900bac4ed0d9222bfbcc56871b9b426ec3503/platform/autoupdate.services/test/unit/src/org/netbeans/modules/autoupdate/services/NewClustersRebootTest.java#L96

the first can be removed when the ant modules are updated https://github.com/apache/netbeans/issues/7871 and I believe we can remove the second occurrence right now - at least tests work locally. Will add another commit and try - some modules have non obvious test invocation relationships.

edit: done, only one usage remains now which is marked by a TODO

**Comment by mbien** [comment]
I forgot about jenkins, the build will likely fail over there :( sorry!

**Comment by lahodaj** [comment]
Thanks!
